packages/uicore/src/components/ScrollContainer/ScrollContainer.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { CSSProperties, FC, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import cx from 'classnames'
+import rafSchd from 'raf-schd'
+import css from './ScrollContainer.css'
+
+export interface ScrollContainerProps {
+  /**
+   * `true`: renders top shadow when required.
+   *
+   * `false`: never renders top shadow.
+   *
+   * default: `true` */
+  showTopShadow?: boolean
+  /**
+   * `true`: renders bottom shadow when required
+   *
+   * `false`: never renders bottom shadow
+   *
+   * default: `true` */
+  showBottomShadow?: boolean
+  /**
+   * height of the container
+   *
+   * default: `100%` */
+  height?: CSSProperties['height']
+  className?: string
+  overflowWrapperClassName?: string
```

Comment:
To keep this more generic, we may want to not expose this and instead consumers can wrap their content in something to add what they need. Cuts down on the chances of consumer-provided styles clashing.

```typescript
<ScrollContainer>
  <div className={css.whateverYouWouldHavePassed}>
    ...
  </div>
</ScrollContainer>

If we do want to keep this, it may make sense to change it to something like `contentClassName` coz `overflowWrapperClassName` is maybe a bit too much of an implementation detail and not obvious what it applies to.

Reply:
Want to keep this element, as the resize observer and scroll event are used on this. Renamed `overflowWrapperClassName` to `contentClassName`.

Reply:
Yeah... deffo keep the element, all I mean is that if someone wants to style that element, they can instead pass a wrapper of their own as part of the `children`. Styling of the content inside your scroll area is their own business. It's fine either way, just a way of minimising the component's API cross-section